### PR TITLE
feat: flow recreate on reboot

### PR DESCRIPTION
**Pull request by discord9**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

recreate flow on flownode on reboot

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
	- Enhanced flow management by integrating a new `FlowMetadataManager` into the `StartCommand` functionality.
	- Introduced a new asynchronous method, `recover_flows`, in `FlownodeBuilder` to handle flow recovery tasks.
	- Updated SQL scripts to support new execution parameters, improving SQL operation configurations.

- **Bug Fixes**
	- Improved error handling by adding a new `ListFlows` variant to the error reporting system.

- **Chores**
	- Adjusted several builder implementations to utilize the new flow metadata manager for better management of flow operations.

### CODE REVIEW

File: src/flow/src/server.rs [hunk 1]
```diff
@@ -296,6 +308,83 @@ impl FlownodeBuilder {
...
     }
 
+    /// recover all flow tasks in this flownode in distributed mode(nodeid is Some(<num>))
+    ///
+    /// or recover all existing flow tasks if in standalone mode(nodeid is None)
+    ///
+    /// TODO(discord9): persisent flow tasks with internal state
+    async fn recover_flows(&self, manager: &FlowWorkerManagerRef) -> Result<usize, Error> {
+        let nodeid = self.opts.node_id;
+        let to_be_recover: Vec<_> = if let Some(nodeid) = nodeid {
```
Comment (waynexia) [inline src/flow/src/server.rs:318]: nit:
```suggestion
        let to_be_recovered: Vec<_> = if let Some(nodeid) = nodeid {
```

File: src/flow/src/server.rs [hunk 2]
```diff
@@ -296,6 +308,83 @@ impl FlownodeBuilder {
...
+                    .await
+                    .map_err(BoxedError::new)
+                    .context(ExternalSnafu)?;
+
+                all_flow_ids.extend(flows.into_iter().map(|(_, id)| id.flow_id()));
+            }
+            all_flow_ids
+        };
+        let cnt = to_be_recover.len();
+        for flow_id in to_be_recover {
```
Comment (waynexia) [inline src/flow/src/server.rs:351]: may consider recovering in parallel in the future

**Review by waynexia — APPROVED** [review]

File: src/flow/src/server.rs [hunk 3]
```diff
@@ -283,6 +288,11 @@ impl FlownodeBuilder {
             self.build_manager(query_engine_factory.query_engine())
                 .await?,
         );
+
+        if let Err(err) = self.recover_flows(&manager).await {
+            common_telemetry::error!("failed to recover flows: {:?}", err);
```
Comment (WenyXu) [inline src/flow/src/server.rs:293]:
```suggestion
            common_telemetry::error!(err; "Failed to recover flows: {:?}");
```

**Comment by killme2008** [comment]
We must be focused on such basic functions. Keep pushing and make the flow better.

**Review by WenyXu — APPROVED** [review]

**Comment by discord9** [comment]
@coderabbitai review

**Pull request merged**